src/Microsoft.Azure.WebJobs.Extensions.Kafka/Output/KafkaProducerEntity.cs

```diff
@@ -22,19 +26,25 @@ public class KafkaProducerEntity
 
         public KafkaAttribute Attribute { get; set; }
 
-        internal async Task SendAndCreateEntityIfNotExistsAsync<T>(T item, Guid functionInstanceId, CancellationToken cancellationToken)
+        internal Task SendAndCreateEntityIfNotExistsAsync<T>(T item, Guid functionInstanceId, CancellationToken cancellationToken)
         {
             var kafkaProducer = this.KafkaProducerFactory.Create(this);
-            if (item is ICollection collection)
+
+            if (item is ICollection)
             {
-                foreach (var collectionItem in collection)
-                {
-                    await kafkaProducer.ProduceAsync(this.Topic, this.GetItemToProduce(collectionItem));
-                }
+                ProduceEvents((ICollection)item, kafkaProducer);
```

Comment:
Is it necessary to make it sync method?

Reply:
We don't want to lose customers who want there events in order so we added this in sync mode.